packages/central-server/src/apiV2/import/importEntities/updateEntities.js

```diff
@@ -130,12 +131,27 @@ export async function updateCountryEntities(
         `Entity code '${code}' is not unique`,
         excelRowNumber,
         'code',
-        countryName,
+        country?.name || NO_COUNTRY,
       );
     }
     codes.push(code);
-    const { parentGeographicalArea, parentEntity } =
-      (await getOrCreateParentEntity(transactingModels, entityObject, country, pushToDhis)) || {};
+    let parentGeographicalArea;
+    let parentEntity;
+
+    // Entities without a country (ie. projects) do not have parent entities
+    if (country) {
```

Comment:
This felt a little hairy, but I think it's a safe enough assumption. Overall this area of the code I find really complex and hard to reason about 😢

Reply:
Yeah lots of 'if' :) No problem to have another one.